src/test/java/igrad/logic/commands/course/CourseAchieveCommandTest.java

```diff
@@ -0,0 +1,100 @@
+import igrad.testutil.CourseInfoBuilder;
+import igrad.testutil.ModuleBuilder;
+
+public class CourseAchieveCommandTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        Optional<Cap> cap = null;
+        assertThrows(NullPointerException.class, () -> new CourseAchieveCommand(cap));
+    }
+
+    @Test
+    public void execute_nullModel_throwsNullPointerException() {
+        Model model = null;
+        Optional<Cap> cap = Optional.of(new Cap("0"));
+
+        CourseAchieveCommand cmd = new CourseAchieveCommand(cap);
+        assertThrows(NullPointerException.class, () -> cmd.execute(model));
+    }
+
+    @Test
+    public void execute_achieveCourse_success() {
+        //set-up our Model
+        Model model = new ModelManager();
+        CourseInfo courseInfo = new CourseInfoBuilder()
+                .withName(VALID_COURSE_NAME_BCOMPSCI)
+                .withSemesters(VALID_COURSE_SEMESTERS_BCOMPSCI)
+                .withCap(VALID_COURSE_CAP_BCOMPSCI)
+                .withCreditsOptional()
+                .build();
+        model.addCourseInfo(courseInfo);
+
+        // set-up expected Model
+        Model expectedModel = new ModelManager();
+        CourseInfo courseInfoCopy = new CourseInfoBuilder()
+                .withName(VALID_COURSE_NAME_BCOMPSCI)
+                .withSemesters(VALID_COURSE_SEMESTERS_BCOMPSCI)
+                .withCap(VALID_COURSE_CAP_BCOMPSCI)
+                .withCreditsOptional()
+                .build();
+        expectedModel.addCourseInfo(courseInfoCopy);
+
+        Optional<Cap> capToAchieve = Optional.of(new Cap(4.5));
+
+        Optional<Cap> estimatedCap = CourseInfo.computeEstimatedCap(model.getCourseInfo(), capToAchieve.get());
+
+        CourseAchieveCommand cmd = new CourseAchieveCommand(capToAchieve);
+
+        String expectedMessage = String.format(MESSAGE_COURSE_ACHIEVE_SUCCESS, estimatedCap.get());
+
+        assertExecuteSuccess(cmd, model, expectedModel, expectedMessage);
+    }
+
+    @Test
+    public void equals() {
+        Optional<Cap> capToAchieve = Optional.of(new Cap(4.5));
+
+        final CourseAchieveCommand courseAchieveCommand = new CourseAchieveCommand(capToAchieve);
+
+        //null
+        assertFalse(courseAchieveCommand.equals(null));
+
+        //same course achieve command
+        assertTrue(courseAchieveCommand.equals(courseAchieveCommand));
+
+        // different type
+        Module module = new ModuleBuilder().build();
+        assertFalse(courseAchieveCommand.equals(module));
+
+        //different course achieve command
+        CourseAchieveCommand otherCourseAchieveCommand;
+        Optional<Cap> otherCapToAchieve = Optional.of(new Cap("0"));
+
+        otherCourseAchieveCommand = new CourseAchieveCommand(otherCapToAchieve);
+
+        //For some weird reason the below returns true. Am not sure of it so am commenting it out first. (Teri)
+        //assertFalse(courseAchieveCommand.equals(otherCourseAchieveCommand));
```

Comment:
ok

Reply:
you need to check `CourseAchieveCommand` whether you  have implemented `equals` correctly